src/utils.js

```diff
@@ -33,17 +35,22 @@ const altWidgetMap = {
     updown: UpDownWidget,
     range: RangeWidget,
     hidden: HiddenWidget,
-  },
-  "date-time": {
-    hidden: HiddenWidget,
   }
 };
 
+const stringFormatWidgets = {
+  "date-time": DateTimeWidget,
+  "email": TextWidget, // XXX: to customize appropriately
+  "hostname": TextWidget,
+  "ipv4": TextWidget,
+  "ipv6": TextWidget,
+  "uri": TextWidget, // XXX: to customize appropriately
```

Comment:
From the specs, interesting input types would be: color, password, search, tel, time